packages/path_provider/path_provider_platform_interface/lib/src/method_channel_path_provider.dart

```diff
@@ -60,19 +59,19 @@ class MethodChannelPathProvider extends PathProviderPlatform {
     if (path == null) {
       return null;
     }
-    return Directory(path);
+    return path;
```

Comment:
No need for all these null checks, just `return path`

Reply:
Nice